projects/hip-tests/catch/unit/occupancy/hipOccupancyAvailableDynamicSMemPerBlock.cc

```diff
@@ -0,0 +1,149 @@
+THE SOFTWARE.
+*/
+
+/**
+ * @brief Returns dynamic shared memory available per block when launching
+ * numBlocks blocks on SM.
+ *
+ * @ingroup Occupancy
+ * Returns in dynamicSmemSize the maximum size of dynamic shared memory
+ * to allow numBlocks blocks per SM.
+ *
+ * @param [out] dynamicSmemSize Returned maximum dynamic shared memory.
+ * @param [in]  f               Kernel function for which occupancy is
+ * calculated.
+ * @param [in]  numBlocks       Number of blocks to fit on SM
+ * @param [in]  blockSize       Size of the block
+ *
+ * @return #hipSuccess, #hipErrorInvalidDevice, #hipErrorInvalidDeviceFunction,
+ * #hipErrorInvalidValue, #hipErrorUnknown
+ */
+
+#include <hip_test_common.hh>
+#include <iomanip>
+#include <iostream>
+
+const size_t LEN = 1024;
+
+__global__ void dynamicReverse(int *d, int n) {
+  __shared__ int s[LEN];
+
+  int t = threadIdx.x;
+  int tr = n - t - 1;
+
+  s[t] = d[t];
+  __syncthreads();
+  d[t] = s[tr];
+}
+
+static __global__ void f1(float *a) { *a = 1.0; }
+
+/**
+ * Test Description
+ * ------------------------
+ *  - This test will verify hipOccupancyAvailableDynamicSMemPerBlock api with
+ *    invalid parameters
+ * Test source
+ * ------------------------
+ * - occupancy/hipOccupancyAvailableDynamicSMemPerBlock.cc
+ * Test requirements
+ * ------------------------
+ * - HIP_VERSION >= 7.1
+ */
+TEST_CASE("Unit_hipOccupancyAvailableDynamicSMemPerBlock_Negative") {
+  size_t dynamicSmemSize;
+  int numBlocks = 1;
+  int blockSize = 1024;
+  SECTION("Number of blocks are zero") {
+    HIP_CHECK_ERROR(hipOccupancyAvailableDynamicSMemPerBlock(&dynamicSmemSize,
+                                                             f1, 0, blockSize),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("Block size is zero") {
+    HIP_CHECK_ERROR(hipOccupancyAvailableDynamicSMemPerBlock(&dynamicSmemSize,
+                                                             f1, numBlocks, 0),
+                    hipErrorInvalidValue);
+  }
+
+#if HT_AMD
+  SECTION("Invalid driver funtion") {
+    HIP_CHECK_ERROR(hipOccupancyAvailableDynamicSMemPerBlock(
+                        &dynamicSmemSize, NULL, numBlocks, blockSize),
+                    hipErrorInvalidDeviceFunction);
+  }
+
+  SECTION("dynamicSmemSize is null") {
+    HIP_CHECK_ERROR(hipOccupancyAvailableDynamicSMemPerBlock(
+                        nullptr, f1, numBlocks, blockSize),
+                    hipErrorInvalidValue);
+  }
+#endif
+}
+
+/**
+ * Test Description
+ * ------------------------
+ *  - This test will verify funtionality of
+ * hipOccupancyAvailableDynamicSMemPerBlock api Test source
+ * ------------------------
+ * - occupancy/hipOccupancyAvailableDynamicSMemPerBlock.cc
+ * Test requirements
+ * ------------------------
+ * - HIP_VERSION >= 7.1
+ */
+TEST_CASE("Unit_hipOccupancyAvailableDynamicSMemPerBlock_Positive") {
+  size_t dynamicSmemSize = 0;
+  int numBlocks = 1;
```

Comment:
We do not need these more constants right, since we already have `LEN`. you might want to rename it a bit better.